pandas/core/reshape/merge.py

```diff
@@ -1028,7 +1028,14 @@ def _validate_specification(self):
                 common_cols = self.left.columns.intersection(
                     self.right.columns)
                 if len(common_cols) == 0:
-                    raise MergeError('No common columns to perform merge on')
+                    raise MergeError('No common columns to perform merge on. '
+                                     'Merge options: right_on={ron}, '
```

Comment:
Can you change the order to be `left_on, right_on, left_index, right_index`, to match the order in the signature?

Reply:
Oh sure I can. will do it.

Reply:
to make this more readable you can break it like
```
raise MergeError(
      'No common ......'   

)
```